pkg/common/patterns.go

```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const EmailPattern = `\b([a-z0-9]{4,25}@[a-zA-Z0-9]{2,12}.[a-zA-Z0-9]{2,6})\b`
+const SubDomainPattern = `\b([0-9a-zA-Z]{2,40})\b`
```

Comment:
This is just alphanumeric. Should it instead be a hostname pattern? (require a valid TLD, etc)